### PR TITLE
Added support for wildcard urls

### LINKED ISSUE #8: Url conventions
The plan is to build and release a basic comparison page based on a url convention

### Url rules
* `/Compare/NServiceBus/4.7.7...5.2.0`
* `/Compare/NServiceBus/4.7...5.2` - Compare max(4.7) vs max (5.2) from a semver perspective.
* `/Compare/NServiceBus/4...5` - Compare max(4) vs max (5) from a semver perspective
* `/Compare/NServiceBus/`   -  compare the 2 latest stable versions from nuget.org

Note: the urls with non deterministic version will do a temp redirect based on the the current version history we have at the moment

### Request for unprocessed nuget

* Setup signalr channel
* Return a page with a spinner
* Queue up a `ProcessNuGetRequest`: Name, LeftVersion, RightVersion
* When done processor publishes `NewComparisonAvailable` / NuGetNotFound / VersionNotFound
* Push those over signalr to notify user

### Plan of attack

- [x] Setup a Azure website: particular-apicomparer-live
- [x] Setup a cname for apicomparer.particular.net
- [x] Setup a web job for the "NuGetProcessor"

### Out of scope for now

* Allow users to register an email if it takes long
* To return json for programmatic comparisons
* Comparing packages that have been splitted

### PR DESCRIPTION
**Pull request by andreasohlund**
Fixes #8 

Outstanding questions:

* How do we test this? Is it worth stubbing out nuget?
* How do we return missing package/version to the user? (Right now we just throw)

### CODE REVIEW

**Comment by andreasohlund** [comment]
@ParticularLabs/apicomparer your thoughts on the questions above?

**Comment by danielmarbach** [comment]
Why do we have to do this on the front-end?

I know when you in the future expand functionality in the frontend we eventually will need to browse packages. Until then I think we should do nuget related stuff in the bakend

**Comment by SeanFeldman** [comment]
I don't like throwing in this case. It is not a logical error that requires an exception.

**Comment by andreasohlund** [comment]
>Why do we have to do this on the front-end?

So we can response redirect easily? (I've missed that from the pull actually)

**Comment by andreasohlund** [comment]
After talking to @danielmarbach we realized that we need to do the checks on the frontend like we do in this pull to avoid sending a message to the backend for each request. This way we can detect and serve static html most of the time

File: APIComparer.Website/CompareModule.cs [hunk 1]
```diff
@@ -12,17 +14,20 @@ public class CompareModule : NancyModule
...
         {
-            Get["/compare/{nugetpackageid}/{leftversion:semver}...{rightversion:semver}"] = ctx =>
+            Get["/compare/{nugetpackageid}/{leftversion}...{rightversion}"] = ctx =>
             {
-                var leftVersion = SemanticVersion.Parse(ctx.leftversion);
-                var rightVersion = SemanticVersion.Parse(ctx.rightversion);
                 var nugetPackageId = (string)ctx.nugetpackageid;
 
+                var leftVersion = ExpandVersionIfNeeded(ctx.leftversion, nugetPackageId, nuGetBrowser);
+                var rightVersion = ExpandVersionIfNeeded(ctx.rightversion, nugetPackageId, nuGetBrowser);
```
Comment (andreasohlund) [inline APIComparer.Website/CompareModule.cs]: Should we response redirect instead if we where able to expand the version? (that would improve caching?)

Ie /somenuget/1.0 => redirects to /somenuget/1.0.4   (given that 1.0.4 was found when expanding)

Comment (SeanFeldman) [inline APIComparer.Website/CompareModule.cs]: +1
Also less confusing (URL and report would match)

**Comment by andreasohlund** [comment]
@ParticularLabs/apicomparer the wildcards now works (with redirects if either the left or right version is expanded).

There are still 2 things left todo:

1. Testing, how should we play that, perhaps just a integration test(s) hitting the real nuget feeds using  https://github.com/NancyFx/Nancy/wiki/Testing-your-application to run the whole module?

2. Error handling: I've raised a separate issue for this and I propose we make that out of scope for this pull? #43 

Comments?

**Comment by andreasohlund** [comment]
Decided to provide some naive error handling for now. Merging

**Pull request merged**